pkg/apiserver/registry/security/workloadidentity/strategy.go

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package workloadidentity
+
+import (
+	"context"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/storage/names"
+
+	"github.com/gardener/gardener/pkg/api"
+	"github.com/gardener/gardener/pkg/apis/security"
+	"github.com/gardener/gardener/pkg/apis/security/validation"
+)
+
+type workloadIdentityStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// Strategy defines the storage strategy for WorkloadIdentity.
+var Strategy = workloadIdentityStrategy{api.Scheme, names.SimpleNameGenerator}
+
+func (workloadIdentityStrategy) NamespaceScoped() bool {
+	return true
+}
+
+func (s workloadIdentityStrategy) PrepareForCreate(_ context.Context, obj runtime.Object) {
+	wi := obj.(*security.WorkloadIdentity)
+	if string(wi.GetUID()) == "" {
+		wi.SetUID(uuid.NewUUID())
+	}
+
+	if wi.GetName() == "" {
+		wi.SetName(s.GenerateName(wi.GetGenerateName()))
+	}
+
+	p, d := validation.GetSubClaimPrefixAndDelimiterFunc()
+	if wi.Status.Sub == "" {
+		wi.Status.Sub = strings.Join([]string{p, wi.GetNamespace(), wi.GetName(), string(wi.GetUID())}, d)
+	}
```

Comment:
Move this to `Canonicalize` to allow removing the above setting of UID/name?

Reply:
Yet, wouldn't it be cleaner to also introduce a strategy for the `/status` subresource?

Reply:
> Move this to Canonicalize to allow removing the above setting of UID/name?

Looking at the [docs](https://github.com/kubernetes/kubernetes/blob/74d578485dcf076fdbbf3c09e1b0ae43c52c19d7/staging/src/k8s.io/apiserver/pkg/registry/rest/create.go#L53-L54), my understanding is that `PrepareForCreate` is suitable place to initialize the status.

> Yet, wouldn't it be cleaner to also introduce a strategy for the /status subresource?

No, we do not need to this because 
- there is no need to have authorization control who/what can update the status
- we do not plan to implement controller to reconcile the resource and update the status. 

The field has the single purpose to provide the client asap with the value of the `sub` claim in the tokens that will be issued on behalf of the WorkloadIdentity.

Reply:
Okay, fair enough. Do we need `Status` at all then? It is a bit confusing/inconsistent to not have a `/status` subresource and no controller that maintains this `status`. Maybe it would be cleaner to move this `Sub` field to `Spec`? For example, the cluster IP in `Service`s is also in `Spec` and maintained by the API server. Just a thought.

Reply:
I was thinking the same, but I am also looking for a field that shows the values were set by the system and not controller by the client (btw, clusterIP can be set by the client). I know status in this resource differs from the usual usage, but as of now I cannot come up with better name.

Reply:
What do you think about renaming `status` to `computedValues`? 

PS. Later, I will add `issuer` field next to `sub` to make it easier for users to get the issuer of the tokens.

Reply:
Hm, not sure. Then let's keep `status`, then we have `spec`+`status` which is at least very familiar to most folks.